InnerEye/ML/lightning_metrics.py

```diff
@@ -258,7 +258,7 @@ def __init__(self) -> None:
 
     def compute(self) -> torch.Tensor:
         preds, targets = self._get_preds_and_targets()
-        return F.binary_cross_entropy(input=preds, target=targets)
+        return F.binary_cross_entropy_with_logits(input=preds, target=targets)
```

Comment:
Actually, there is a problem here. Line 304 of `lightning_models.py` computes the posteriors from the logits. Here, you are now again treating the incoming numbers as logits. It makes the warning go away, but the results will be off. And I think that's also where the test failure comes from.

Reply:
We've been discussing how to best resolve that. My suggestion would be:
- Add a flag to `ScalarMetricsBase` that indicates if the metric expects logits or posteriors. By default, it wants posteriors, but `BinaryCrossEntropyWithLogits` wants logits.
- In `compute_and_log_metrics`, you check that flag, and pass in either the logits or the posteriors.

Reply:
Addressed. This fixed the failing test. Thanks!